main.cpp

```diff
@@ -58,17 +122,24 @@ struct List {
     }
 
     int pop_front() {
-        int ret = head->value;
-        head = head->next;
-        return ret;
+        if (head != nullptr)
+        {
+			int ret = head->value;
+			head = std::move(head->next);
+			return ret;
+
+        }
     }
 
     void push_front(int value) {
-        auto node = std::make_shared<Node>(value);
-        node->next = head;
+        auto node = std::unique_ptr<Node>(new Node(value));
```

Comment:
```suggestion
        auto node = std::make_unique<Node>(value);
```
C++14引入的make_unique：异常安全&简单明了。

Reply:
嗯，好的，以后就都用mak_unique；空格的问题，在github上就发现排版很诡异，但是没想到是tab和空格混用造成的，以为是github的显示问题，以后多注意。